internal/future/storage/redisstore/redis.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package storage
+package redisstore
```

Comment:
I wouldn't create a separate package. Keep everything in 1 package, `storage`.
See how `package monitoring` was done. Since all the different modes are in 1 package you can have a very tight exported API (it's just 1 method, `Setup(cfg config.View, mux *http.ServeMux)`)
If you have separate packages then redis, etc are exposed outside of the package which isn't desirable.